src/components/Documents/DocumentTable.jsx

```diff
@@ -32,47 +32,166 @@ import { EmptyListNotification, LoadingAnimation } from '../Notification';
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center'
+    },
+    {
+      headerName: 'Upload Date',
+      field: 'upload date',
+      minWidth: 120,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center'
+    },
+    {
+      headerName: 'Expiration Date',
+      field: 'expiration date',
+      minWidth: 120,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center'
+    },
+    {
+      headerName: 'Preview File',
+      field: 'preview file',
+      minWidth: 100,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      sortable: false,
+      filterable: false,
+      renderCell: (fileUrl) => (
+        <GridActionsCellItem
+          key={String(fileUrl)}
+          icon={<FileOpenIcon />}
+          onClick={() => handleShowDocumentLocal(fileUrl)}
+          label="Preview"
+        />
+      )
+    },
+    {
+      headerName: 'Sharing',
+      field: 'sharing',
+      type: 'actions',
+      minWidth: 80,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      getActions: (data) => [
+        <GridActionsCellItem
+          icon={<ShareIcon />}
+          onClick={() => handleAclPermissionsModal('documents', data.name, data.type)}
+          label="Share"
+        />
+      ]
+    },
+    {
+      headerName: 'Delete',
+      field: 'actions',
+      type: 'actions',
+      minWidth: 80,
+      flex: 1,
+      headerAlign: 'center',
+      align: 'center',
+      getActions: (data) => [
+        <GridActionsCellItem
+          icon={<DeleteOutlineOutlinedIcon />}
+          onClick={() => handleSelectDeleteDoc(data.row.delete)}
+          label="Delete"
+        />
+      ]
+    }
   ];
 
   const determineDocumentsTable = documentListObject?.docList?.length ? (
     // render if documents
-    <ThemeProvider theme={theme}>
-      <Container>
-        <TableContainer component={Paper} sx={{ margin: '1rem 0' }}>
-          <Table aria-label="Documents Table">
-            <TableHead>
-              <TableRow>
-                {columnTitlesArray.map((columnTitle) => (
-                  <TableCell key={columnTitle} align="center">
-                    {columnTitle}
-                  </TableCell>
-                ))}
-              </TableRow>
-            </TableHead>
-            <TableBody>
-              {documentListObject?.docList.map((document) => (
-                <DocumentTableRow
-                  key={document.name}
-                  document={document}
-                  handleAclPermissionsModal={handleAclPermissionsModal}
-                  handleSelectDeleteDoc={handleSelectDeleteDoc}
-                />
-              ))}
-            </TableBody>
-          </Table>
-        </TableContainer>
-      </Container>
-    </ThemeProvider>
+    <Container>
+      <DataGrid
+        columns={columnTitlesArray}
+        rows={documentListObject?.docList.map((document) => ({
+          id: document.name,
+          type: document.type,
```

Comment:
For the document type, think we could find a way to return a formatted string using `src/constants/doc_types.js`.

So something along the lines of `type: DOC_TYPES[document.type]` with DOC_TYPES imported.

Of course, it would look like there has been adjustments to how `document.type` was pushed into Solid from previous PRs so not all the types are consistently camelCase:

![Screen Shot 2023-11-19 at 15 28 46](https://github.com/codeforpdx/PASS/assets/14917816/e22e1f3a-b83a-48ad-8dc4-95e63be7e89c)

But I think we could reset everything by clearing whatever document list you're currently testing with by deleting the documents and adding them again with the add documents button.

Reply:
Updated to use DOC_TYPES.